src/fheroes2/agg/agg_image.cpp

```diff
@@ -4126,10 +4126,17 @@ namespace fheroes2
                     const Sprite & originalImage = GetICN( ICN::ARTIFACT, 83 );
                     SubpixelResize( originalImage, _icnVsSprite[id][82] );
                 }
+                else if ( _icnVsSprite[id].size() > 63 ) {
```

Comment:
This one should be just `if` without `else`: we just check that we are going to fix image that exists and we do want to fix both images.
```suggestion
                if ( _icnVsSprite[id].size() > 63 ) {
```